cmd/otelcol/main.go

```diff
@@ -25,14 +26,19 @@ import (
 )
 
 func main() {
-	handleErr := func(message string, err error) {
-		if err != nil {
-			log.Fatalf("%s: %v", message, err)
-		}
+	params, err := applicationParameters()
+	if err != nil {
+		log.Fatal(err)
 	}
 
+	run(params)
+}
+
+func applicationParameters() (service.Parameters, error) {
 	factories, err := defaultcomponents.Components()
-	handleErr("Failed to build default components", err)
+	if err != nil {
+		return service.Parameters{}, fmt.Errorf("failed to build default components: %v", err)
```

Comment:
nit: It's cleaner to use `errors.Wrap("some message", err)` to propagate errors, see https://godoc.org/github.com/pkg/errors#Wrap.
Applies to all other places where you propagate error messages.